pkg/api/controller.go

```diff
@@ -631,6 +652,53 @@ func (c *Controller) StsLogin(w http.ResponseWriter, r *http.Request, body apige
 	writeResponse(w, r, http.StatusOK, responseToken)
 }
 
+func (c *Controller) verifyLinkAddress(repository, branch, path, physicalAddress string) error {
+	address, signature, found := strings.Cut(physicalAddress, ",")
+	if !found {
+		return fmt.Errorf("address is not signed: %w", graveler.ErrLinkAddressInvalid)
+	}
+
+	stringToVerify := getAddressToSign(repository, branch, path, address)
+	decodedSig, err := base64.URLEncoding.DecodeString(signature)
+	if err != nil {
+		return fmt.Errorf("malformed address signature: %s: %w", stringToVerify, graveler.ErrLinkAddressInvalid)
+	}
+
+	h := hmac.New(sha256.New, []byte(c.Config.Blockstore.Encrypt.SecretKey))
+	h.Write([]byte(stringToVerify))
+	calculated := h.Sum(nil)
```

Comment:
I think this can and should use encryptAddress() (or whatever it's new name will be).